<a href="https://colab.research.google.com/github/n8mcdunna/DS-Unit-1-Sprint-2-Statistics/blob/master/module3/Nathan_Confidence_Intervals_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Confidence Intervals

Soft drinks like Coke and Pepsi are manufactured to have a standard caffeine content. For example, a 12-oz serving of Coke has 34mg of caffeine, and a 12-oz serving of Pepsi has 37.6mg of caffeine. However, fountain soft drinks are typically mixed in individual restaurant dispensers, so it is more difficult to maintain a standard level of caffeine per serving. In this study, researchers randomly sampled Coke, Diet Coke, Pepsi, and Diet Pepsi at a set of franchise restaurants and measured the caffeine content in 12oz of each soft drink2. The data is found in the Soda.xlsx dataset.

Because individuals can be sensitive to caffeine – and because the manufacturers are interested in product consistency – we wish to estimate the mean caffeine content in 12oz of Coke served in franchise restaurants using a 95% confidence interval. 

You can find the Coke data here: 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv'

The first variable is the sample ID and the second variable is the caffeine content in the 12-oz sample measured in mg.

Source: A.N. Garand and L.N. Bell (1997). "Caffeine Content of Fountain and Private-Label Store Brand Carbonated Beverages," Journal of the American Dietetic Association, Vol. 97, #2, pp. 179-182.




###1) Load the dataset and print the first few rows

In [3]:
import pandas as pd
import numpy as np
from scipy.stats import t

In [4]:
df = pd.read_csv('https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv')
df.head()

Drink  Caffeine
0      1     47.32
1      2     43.78
2      3     48.12
3      4     43.25
4      5     46.42

###2) Calculate the mean, SD, SE caffeiene content and n for the sample.  Summarize your results in a sentence or two.

In [6]:
mean = df['Caffeine'].mean()
print('mean: ', mean)
sd = df['Caffeine'].std()
print('standard deviation: ', sd)
n = df['Caffeine'].count()
print("n: ", n)
se = sd / n ** 0.5
print("se: ", se)

mean:  37.9402
standard deviatin:  5.243756828216712
n:  50
se:  0.7415792024250598


###Summary
The sample mean is higher than both the pepsi and coke serving size examles. There are greater than 30 samples so the t disribuition should look like a normal distribution. 

###3) Find t* for a 95% confidence interval.  

Use the starter code below and fill in the degrees of freedom.

In [7]:
from scipy.stats import t

#Don't worry too much about where the 0.975 comes from.  It has to do
#with wanting to determine the *middle* 95% of the t-distribution
#We're going to learn
#how to calculate a 95% CI this easy way in just a minute.

#Recall that n = 223 for the body temp problem.
t_star = t.ppf(0.975,df= (n-1))
print('t_star =', t_star)

t_star = 2.009575234489209


###4) Calculate the margin of error for a 95% confidence interval for the mean caffeine content in a 12-oz Coke.



In [10]:
margin_of_error = t_star*se
margin_of_error

1.49025919960566

###5) Calculate a 95% CI for the mean caffeine content in a 12-oz fountain Coke with the CI formula using the summary statistics and t* that you calculated above.

In [14]:
lower_bound = mean - margin_of_error
upper_bound = mean + margin_of_error
print(lower_bound, upper_bound)

36.44994080039434 39.43045919960566


###6) Calculate a 95% CI for the mean caffeiene content in a 12-oz fountain Coke using the t-interval function in Python.

In [12]:
t.interval(alpha= 0.95, df= (n-1), loc= mean, scale= se)

(36.44994080039434, 39.43045919960566)

###7) Compare the two confidence intervals you calculated.  Do they match?  Should they?

The answers match as they should since the t-interval function in python uses the confidence interval formula and summary statistics to calculate the range.

###8) Interpret the meaning of the 95% confidence interval for the mean caffeiene content in a 12-oz fountain Coke. in a sentence or two.

We are 95% confident that the population mean for caffeine in a 12 oz fountain coke is between 36.4 and 39.4 grams.

###9) Using the t-interval Python function, calculate a 90% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [15]:
t.interval(alpha= 0.90, df= (n-1), loc= mean, scale= se)

(36.696904726749196, 39.1834952732508)

We are 90% confident that the population mean for caffeine in a 12 oz fountain coke is between 36.69 and 39.18 grams.

###10) Using the t-interval Python function, calculate a 99% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [16]:
t.interval(alpha= 0.99, df= (n-1), loc= mean, scale= se)


(35.95280335285685, 39.92759664714315)

We are 99% confident that the population mean for caffeine in a 12 oz fountain coke is between 35.9 and 39.9 grams.

## Stretch goals:

###1) The correspondence between confidence intervals and hypothesis tests.

Read [this](https://https://blog.minitab.com/blog/adventures-in-statistics-2/understanding-hypothesis-tests-confidence-intervals-and-confidence-levels#:~:text=If%20a%20hypothesis%20test%20produces,corresponding%20confidence%20level%20is%2095%25.&text=If%20the%20confidence%20interval%20does,the%20results%20are%20statistically%20significant.) article about the correspondence between confidence intervals and hypothesis tests.  Feel free to read the whole article, but the relevant part can be found under the heading Why P Values and Confidence Intervals Always Agree About Statistical Significance.

Imagine you work for quality control at Coke and are tasked with making sure that the caffeiene content in the fountain beverages served in restaurants is the same as in a 12-oz can of Coke (34mg).  If you believe that the mean caffeiene content in fountain coke is not 34mg, you must re-train the franchise managers to make sure the Coke served has the correct caffeiene level.

Based on the confidence interval you calculated in the assignment, do you believe that the mean caffeiene content is statistically significantly different from 34 mg in a 12-oz serving?


Yes, because the lower bound in the 99% confidence interval (35.9) is well above the what the population mean should be of 34.

###2) If we increased the sample size from 50 to 100 but the sample mean and SD remained the same, describe **two** ways the margin of error would change.  Would the margin of error become smaller or larger?

Answer to what would change here.
The margin of error would become smaller as the sample size increased.


In [21]:
n2= 100
t_star2 = t.ppf(0.975,df= (n2-1))
se2 = sd / n2 ** 0.5
margin_of_error2 = t_star2*se2
print("margin of error for 100 samples: ",margin_of_error2)
print("margin of error for 50 samples: ", margin_of_error)

margin of error for 100 samples:  1.0404751188137005
margin of error for 50 samples:  1.49025919960566
